addon/Feeds/PlacesStatsFeed.js

```diff
@@ -15,6 +17,21 @@ module.exports = class PlacesStatsFeed extends Feed {
       PlacesStatsUpdate(historySize, bookmarksSize)
     ));
   }
+  sendStatsPing() {
+    this.options.metadataStore.asyncGetOldestInsert().then(timestamp => (
+      Promise.all([
+        PlacesProvider.links.getHistorySizeSince(timestamp),
+        this.options.metadataStore.asyncCountAllItems()
```

Comment:
asyncCountAllItems() might return null as well. Instead of returning nulls from metadata store, consider just raising exceptions from there and letting the client code decide how to handle those exceptions.